src/main/java/levy/art/RijkSearchFrame.java

```diff
@@ -0,0 +1,131 @@
+package levy.art;
+
+import javax.imageio.ImageIO;
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import java.io.IOException;
+import java.net.URL;
+
+import com.andrewoid.ApiKey;
+import hu.akarnokd.rxjava3.swing.SwingSchedulers;
+import io.reactivex.rxjava3.disposables.Disposable;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+import levy.art.json.ArtObject;
+import levy.art.json.RijksCollection;
+
+
+public class RijkSearchFrame extends JFrame {
+    private JTextField searchField;
+    private JButton prevButton;
+    private JButton nextButton;
+    private JPanel imagesPanel;
+    private int currentPage = 1;
+    private RijkService rijkService;
+    private Disposable disposable;
+    private String apiKey;
+
+    public RijkSearchFrame(RijkService rijkService) {
+        this.rijkService = rijkService;
+
+        ApiKey apiKey = new ApiKey();
+        this.apiKey = apiKey.get();
+
+        setTitle("Rijks");
+        setSize(1000, 800);
+        setDefaultCloseOperation(EXIT_ON_CLOSE);
+        setLayout(new BorderLayout());
+
+        searchField = new JTextField(20);
+        prevButton = new JButton("Previous Page");
+        nextButton = new JButton("Next Page");
+        JPanel topPanel = new JPanel();
+        topPanel.add(prevButton);
+        topPanel.add(searchField);
+        topPanel.add(nextButton);
+        add(topPanel, BorderLayout.NORTH);
+
+        imagesPanel = new JPanel(new GridLayout(3, 4, 10, 10));
+        add(new JScrollPane(imagesPanel), BorderLayout.CENTER);
+
+        prevButton.addActionListener(e -> {
+            if (currentPage > 1) {
+                currentPage--;
+                searchArt();
+            }
+        });
+
+        nextButton.addActionListener(e -> {
+            currentPage++;
+            searchArt();
+        });
+
+        searchField.addActionListener(e -> {
+            currentPage = 1;
+            searchArt();
+        });
+
+        searchArt();
+    }
+
+    private void searchArt() {
+        if (disposable != null && !disposable.isDisposed()) {
+            disposable.dispose();
+        }
+
+        ApiKey apiKey = new ApiKey();
+        String keyString = apiKey.get();
```

Comment:
You are already doing this on line 32, you don't need to do it here as well

Reply:
<img width="817" alt="Screenshot 2024-05-27 at 7 04 02 PM" src="https://github.com/YaffaLevy/levy-artMuseum-2024/assets/135287752/0c1e902a-73d9-49a9-a420-5c0a5fb7cf14">

Reply:
i fixed it is this better?

Reply:
?

Reply:
Yes, this has to be there all 3 times. That's how you pass the API key to the service.